server/vehicle-persistence.lua

```diff
@@ -121,4 +132,50 @@ AddEventHandler('entityRemoved', function(entity)
         local passenger = passengers[i]
         SetPedIntoVehicle(passenger.ped, veh, passenger.seat)
     end
+end)
+
+if persistence == 1 then return end
+
+local function checkVehicleExist(plate)
+    local vehicles = GetGamePool('CVehicle')
+    for i = 1, #vehicles do
+        local vehicle = vehicles[i]
+        if qbx.getVehiclePlate(vehicle) == plate then
+            return true
+        end
+    end
+end
+
+local function spawnVehicle(coords, id, model, props)
+    local _, veh = qbx.spawnVehicle({
+        spawnSource = vector4(coords.x, coords.y, coords.z, coords.w),
+        model = model,
+        props = props
+    })
+    exports.qbx_core:EnablePersistence(veh)
+    Entity(veh).state:set('vehicleid', id, false)
+    SetVehicleDoorsLocked(veh, 2)
+    TriggerClientEvent('qbx_core:client:removeVehZone', -1, id)
+end
+
+lib.callback.register('qbx_core:server:getVehiclesToSpawn', function()
+    local vehicles = {}
+    local query = 'SELECT id, plate, coords FROM player_vehicles WHERE state = 0'
+    local results = MySQL.query.await(query)
+    for _, data in pairs(results) do
+        local coords = json.decode(data.coords)
+        if coords and not checkVehicleExist(data.plate) then
+            vehicles[#vehicles + 1] = {
+                id = data.id,
+                coords = coords,
+            }
+        end
+    end
+    return vehicles
+end)
+
+RegisterNetEvent('qbx_core:server:spawnVehicle', function(id, coords)
+    local vehicle = exports.qbx_vehicles:GetPlayerVehicle(id)
+    if not vehicle then return end
+    spawnVehicle(coords, id, vehicle.modelName, vehicle.props)
```

Comment:
This is an exploitable event. Needs a server side check. Once a cache exists, you can use it to verify that the id vehicle exists in cache near the specified coords